tests/nlu/classifiers/test_diet_classifier.py

```diff
@@ -49,6 +50,116 @@
+            config={**DIETClassifierGraphComponent.get_default_config(), **config},
+            model_storage=default_model_storage,
+            execution_context=default_execution_context,
+            resource=default_diet_resource,
+        )
+
+    return inner
+
+
+@pytest.fixture()
+def create_train_load_and_process_diet(
+    nlu_data_path: Text,
+    create_diet: Callable[..., DIETClassifierGraphComponent],
+    default_model_storage: ModelStorage,
+    default_execution_context: ExecutionContext,
+    default_diet_resource: Resource,
+    train_load_and_process_diet: Callable[..., Message],
+) -> Callable[..., Message]:
+    def inner(
+        diet_config: Dict[Text, Any],
+        pipeline: Optional[List[Dict[Text, Any]]] = None,
+        training_data: str = nlu_data_path,
+        message_text: Text = "Rasa is great!",
+        expect_intent: bool = True,
+    ) -> Message:
+        diet = create_diet(diet_config)
+        return train_load_and_process_diet(
+            diet=diet,
+            pipeline=pipeline,
+            training_data=training_data,
+            message_text=message_text,
+            expect_intent=expect_intent,
+        )
+
+    return inner
+
+
+@pytest.fixture()
+def train_load_and_process_diet(
+    nlu_data_path: Text,
+    create_diet: Callable[..., DIETClassifierGraphComponent],
+    default_model_storage: ModelStorage,
+    default_execution_context: ExecutionContext,
+    default_diet_resource: Resource,
+) -> Callable[..., Message]:
+    def inner(
+        diet: DIETClassifierGraphComponent,
+        pipeline: Optional[List[Dict[Text, Any]]] = None,
+        training_data: str = nlu_data_path,
+        message_text: Text = "Rasa is great!",
+        expect_intent: bool = True,
+    ) -> Message:
+
+        if not pipeline:
+            pipeline = [
+                {"name": "WhitespaceTokenizer"},
+                {"name": "CountVectorsFeaturizer"},
+            ]
+
+        loaded_pipeline = [
+            registry.get_component_class(component.pop("name"))(component)
+            for component in copy.deepcopy(pipeline)
+        ]
+
+        importer = RasaFileImporter(training_data_paths=[training_data])
+        training_data = importer.get_nlu_data()
+
+        for component in loaded_pipeline:
+            component.train(training_data)
+
+        diet.train(training_data=training_data)
+
+        message = Message(data={TEXT: message_text})
+        for component in loaded_pipeline:
+            component.process(message)
+
+        message2 = copy.deepcopy(message)
+
+        classified_message = diet.process([message])[0]
+
+        if expect_intent:
+            assert classified_message.data["intent"]["name"]
+
+        loaded_diet = create_diet(diet.component_config, load=True)
+
+        classified_message2 = loaded_diet.process([message2])[0]
+
+        assert classified_message2.fingerprint() == classified_message.fingerprint()
+        return classified_message
```

Comment:
how about making this all about training the classifier moving the asserting of a message (and possible the processing of the message) to a separate function?

Reply:
hmm but where would the loading go? because it currently checks that the persisting and loading does not change the prediction at all.

Reply:
Also I'm worried adding more fixture functions will make it harder to follow, given they all have to have `inner` to have access to the fixtures... I think this structure works well for all the tests now - and maybe we can re-asses in the future if it is inflexible for whatever reason?

Reply:
mhm, I imagined something like the snippet below. In my opinion that would make the single unit tests more verbose but easier to read as you don't have to jump through multiple fixtures in order to understand what's going on. However, I've seen that it was more or less done similarly before so fine to me.

```python
classifier = create_diet(...)
path = train_diet(classifier, pipeline, ...)
loaded_classifier = load_diet(...)

prediction1 = classifier.process(...)
prediction2 = classifier.process(...)
```

Reply:
The problem is that you need to pre-process the messages before training and prediction. And for prediction you need to re-use the featurizors that were trained before. So there would be a lot of passing around things like pipelines.

Reply:
yeah, then skip it 👍🏻